flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -909,7 +919,21 @@ private int handleError(Throwable t) {
 		}
 		LOG.error("Error while running the command.", t);
 
-		t.printStackTrace();
+		// check if the error was due to an invalid program in detached mode.
+		if (t instanceof ProgramInvocationException && t.getCause() instanceof DetachedProgramException) {
+			System.err.println(t.getCause().getMessage());
+			// now trace to the user's main method. We don't wanna show unnecessary information
+			// in this particular case.
```

Comment:
True. I didn't wanna change the existing stack traces.
This will require maintaining a global variable for whether the program was run in interactive mode. The `DetachedProgramException` is a special case where we don't wanna clutter the stack trace with flink's trace, instead just show the user program's stack trace.
In other cases, this might not be the case.